core-api/src/main/java/com/optimizely/ab/bucketing/DecisionService.java

```diff
@@ -131,7 +131,13 @@ public DecisionService(@Nonnull Bucketer bucketer,
         if (ExperimentUtils.isUserInExperiment(projectConfig, experiment, filteredAttributes)) {
             String bucketingId = userId;
             if (filteredAttributes.containsKey(ControlAttribute.BUCKETING_ATTRIBUTE.toString())) {
-                bucketingId = filteredAttributes.get(ControlAttribute.BUCKETING_ATTRIBUTE.toString());
+                if (String.class.isInstance(filteredAttributes.get(ControlAttribute.BUCKETING_ATTRIBUTE.toString()))) {
+                    bucketingId = (String) filteredAttributes.get(ControlAttribute.BUCKETING_ATTRIBUTE.toString());
+                    logger.info("bucketingId is valid");
+                } 
+                else {
+                    logger.warn("bucketingID has type mismatch, defaulted to userId ");
```

Comment:
nit: This should just say `BucketingID attribute is not a string. Defaulted to userId.`